### PR TITLE
fix: mysql provider did not implement storage.Provider

### PR DESCRIPTION
**Pull request by llorllale**
* fix: mysql provider did not implement storage.Provider (missing CreateStore(string) error method)
* fix: removed mysql.Provider.db attribute because it is not needed
* fix: mysql storage setup: tests were not actually run because `waitForSQLDBToStart()` failed and then the tests exited with code `0`

Signed-off-by: George Aristy <george.aristy@securekey.com>

### CODE REVIEW

**Comment by llorllale** [comment]
Test coverage fails due to unreachable code.

File: pkg/storage/mysql/mysqlstore.go [hunk 1]
```diff
@@ -55,28 +54,86 @@ const (
 	useDBQuery                = "USE "
 )
 
-// NewProvider instantiates Provider
-func NewProvider(dbPath string, opts ...Option) (*Provider, error) {
+// NewProvider instantiates Provider.
+// Example DB Path root:my-secret-pw@tcp(127.0.0.1:3306)/.
+func NewProvider(dbPath string, opts ...Option) (p *Provider, err error) {
```
Comment (Baha-sk) [inline pkg/storage/mysql/mysqlstore.go:59]: Why use named returns and returning `prov` at the end? This is misleading.

Comment (llorllale) [reply pkg/storage/mysql/mysqlstore.go:59]: @Baha-sk return params were named in order to wrap any error that might have been thrown with another error that may be thrown while attempting to close the db

Comment (llorllale) [reply pkg/storage/mysql/mysqlstore.go:59]: @Baha-sk I've modified the code to `return p, err` - please check

File: pkg/storage/mysql/mysqlstore.go [hunk 2]
```diff
@@ -55,28 +54,86 @@ const (
...
+
+	db, err := sql.Open("mysql", p.dbURL)
+	if err != nil {
+		return fmt.Errorf("failed to open connection: %w", err)
+	}
+
+	defer func() {
+		closeErr := db.Close()
+		if closeErr != nil {
+			err = fmt.Errorf("%s : failed to close db connection : %w", err, closeErr)
```
Comment (DRK3) [inline pkg/storage/mysql/mysqlstore.go:110]: extra space before colon

File: pkg/storage/mysql/mysqlstore.go [hunk 3]
```diff
@@ -55,28 +54,86 @@ const (
...
 	if err != nil {
-		return nil, fmt.Errorf("failed to open connection: %w", err)
+		return nil, fmt.Errorf("failed to open MySQL using url %s : %w", dbPath, err)
 	}
 
-	p := &Provider{
+	defer func() {
+		closeErr := db.Close()
+		if closeErr != nil {
+			err = fmt.Errorf("%s : failed to close db connection : %w", err, closeErr)
```
Comment (DRK3) [inline pkg/storage/mysql/mysqlstore.go:72]: extra space before colon

File: pkg/storage/mysql/mysqlstore.go [hunk 4]
```diff
@@ -55,28 +54,86 @@ const (
...
+		}
+	}()
+
+	err = db.Ping()
+	if err != nil {
+		return nil, fmt.Errorf("failed to ping MySQL at url %s : %w", dbPath, err)
```
Comment (DRK3) [inline pkg/storage/mysql/mysqlstore.go:78]: extra space before colon

**Pull request merged**